cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -233,18 +237,27 @@ impl ABIMachineSpec for AArch64MachineDeps {
             if let Some(param) = try_fill_baldrdash_reg(call_conv, param) {
                 assert!(rc == RegClass::I64);
                 ret.push(param);
+            } else if let ir::ArgumentPurpose::StructArgument(size) = param.purpose {
+                let offset = next_stack as i64;
+                let size = size as u64;
+                next_stack += size;
```

Comment:
Such an invariant shouldn't exist. Or at least cg_clif doesn't honour it. Maybe the abi requires it, in which case it would be nice to automatically add padding.

Reply:
My reading of the [aarch64 spec](https://github.com/ARM-software/abi-aa/blob/master/aapcs64/aapcs64.rst#parameter-passing-rules) is that on-stack args have 8-byte-aligned sizes; as far as I can tell the [x64 SystemV spec](https://www.uclibc.org/docs/psABI-x86_64.pdf) also shows on-stack args broken down into "eightbytes". I've added an align-to-8 assert for both architectures.